db/db_creator.py

```diff
@@ -41,33 +41,60 @@ def __init__(
         )
         self.engine: Engine = create_engine(self.conn_str)
 
-        self.append_table_names: List[str] = append_table_names
-        self.append_table_names += DEFAULT_APPEND_TABLE_NAMES
-
     def get_table_names(self) -> List[str]:
+        '''
+        Gets table names using the SQLAlchemy Engine object.
+        '''
         logger.debug('get_table_names')
         return self.engine.table_names()
 
     def migrate(self) -> DBCreator:
+        '''
+        Updates database schema using yoyo-migrations and the migration files in the
+        migrations directory collocated with this file (db_creator.py).
+        '''
         logger.debug('migrate')
         backend = get_backend(self.conn_str)
         migrations = read_migrations(MIGRATIONS_PATH)
         with backend.lock():
             backend.apply_migrations(backend.to_apply(migrations))
         return self
 
-    def drop_records(self) -> DBCreator:
+    def drop_records(self, spec_table_names: Union[Sequence[str], None] = None) -> DBCreator:
+        '''
+        Drops records from either the specified database tables or all of the
+        application-managed database tables; tables managed by yoyo-migrations are ignored.
+        '''
         logger.debug('drop_records')
+        app_table_names = [
+            table_name for table_name in self.get_table_names() if 'yoyo' not in table_name
+        ]
+        logger.debug(f'app_table_names: {app_table_names}')
+
+        if spec_table_names is None:
```

Comment:
Can we have a log info/debug message saying this case is entered? similarly for the else case. Info if it worth tracking in the prod logs

Reply:
Sure, I could do that.

Reply:
I added some `INFO`-level logging as you requested, so I'm resolving this. See [this commit](https://github.com/tl-its-umich-edu/course-inventory/pull/129/commits/96079b7a94d06d5ba7efcd3cd0e3ca3f09de10d7).